jvm/core/src/main/java/ml/dmlc/tvm/contrib/GraphModule.java

```diff
@@ -0,0 +1,170 @@
+package ml.dmlc.tvm.contrib;
+
+import ml.dmlc.tvm.Function;
+import ml.dmlc.tvm.Module;
+import ml.dmlc.tvm.NDArray;
+import ml.dmlc.tvm.TVMContext;
+
+/**
+ * Wrapper runtime module.
+ * This is a thin wrapper of the underlying TVM module.
+ * you can also directly call set_input, run, and get_output
+ * of underlying module functions.
+ */
+public class GraphModule {
+  private Module module;
+  private TVMContext ctx;
+
+  private Function fsetInput;
+  private Function frun;
+  private Function fgetOutput;
+  private Function fgetInput;
+  private Function fdebugGetOutput;
+  private Function floadParams;
+
+  GraphModule(Module module, TVMContext ctx) {
+    this.module = module;
+    this.ctx = ctx;
+    fsetInput = module.getFunction("set_input");
+    frun = module.getFunction("run");
+    fgetInput = module.getFunction("get_input");
+    fgetOutput = module.getFunction("get_output");
+    try {
+      fdebugGetOutput = module.getFunction("debug_get_output");
+    } catch (IllegalArgumentException ignored) {
+      // ignore
+    }
+    floadParams = module.getFunction("load_params");
+  }
+
+  /**
+   * Release the GraphModule.
+   * <p>
+   * We highly recommend you to do this manually since the GC strategy is lazy.
+   * </p>
+   */
+  public void release() {
+    fsetInput.release();
+    frun.release();
+    fgetInput.release();
+    fgetOutput.release();
+    if (fdebugGetOutput != null) {
+      fdebugGetOutput.release();
+    }
+    floadParams.release();
+    module.release();
+  }
+
+  /**
+   * Set inputs to the module.
+   * @param key The input key.
+   * @param value The input value
+   * @return self.
+   */
+  public GraphModule setInput(String key, NDArray value) {
+    NDArray input = value;
+    if (!value.ctx().equals(ctx)) {
+      input = NDArray.empty(value.shape(), ctx);
+      value.copyTo(input);
+    }
+    fsetInput.pushArg(key).pushArg(input).invoke();
+    return this;
+  }
+
+  /**
+   * Set inputs to the module
+   * @param key The input key.
+   * @param value The input value.
+   * @return self.
+   */
+  public GraphModule setInput(int key, NDArray value) {
```

Comment:
I think batched inputs can still be described as NDArray.

Reply:
But here we only supports a single input key. We may want a map for multiple inputs.

Reply:
you can do `g.setInput(k1, v1).setInput(k2, v2).setInput...`

Reply:
Yes. User can always set input one by one. I'm thinking whether we can make the API simpler. Maybe we can do this later if it requires significant extra work.

Reply:
I prefer not to provide another two functions `setInput(Map<String, NDArray>)` and `setInput(Map<Integer, NDArray>)` since now users need to construct `Map` using `Map.put(k, v)` first - which is actually less convenient.